src/framework/mpas_timekeeping.F

```diff
@@ -810,7 +822,94 @@ subroutine mpas_print_alarm(clock, alarmID, ierr)
    end subroutine mpas_print_alarm
 
 
-   logical function mpas_is_alarm_ringing(clock, alarmID, interval, ierr)
+      !-----------------------------------------------------------------------
+      !  function mpas_is_alarm_active
+      !
+      !> \brief Determine if an alarm is currently active.
+      !> \author Andy Stokely
+      !> \date   10/01/2025
+      !> \details This function checks whether the current clock time falls
+      !>   within the active start and stop times of the given alarm. If so,
+      !>   it returns `.true.`. An optional error code may also be returned.
+      !-----------------------------------------------------------------------
+      logical function mpas_is_alarm_active(clock, alarm, ierr)
+         implicit none
+         type(MPAS_Clock_type), intent(in) :: clock
+         type(MPAS_Alarm_type), pointer    :: alarm
+         integer, intent(out), optional    :: ierr
+         type(MPAS_Time_type)              :: now
+
+         if (present(ierr)) ierr = 0
+
+         now = mpas_get_clock_time(clock, MPAS_NOW, ierr=ierr)
+         mpas_is_alarm_active = mpas_time_in_interval( &
+               alarm%activeStartTime, alarm%activeStopTime, now)
+
+      end function mpas_is_alarm_active
+
+
+      !-----------------------------------------------------------------------
+      !  function mpas_prev_ring_in_window
+      !
+      !> \brief Check if the alarm’s previous ring was inside its window.
+      !> \author Andy Stokely
+      !> \date   10/01/2025
+      !> \details This function tests whether the alarm’s `prevRingTime`
+      !>   occurred strictly within the defined start and stop times of the
+      !>   alarm’s active window. The check uses an open interval:
+      !>
+      !>       (start, stop)
+      !>
+      !>   The boundaries themselves are excluded. If the previous ring time
+      !>   lies inside this open interval, the function returns `.true.`.
+      !-----------------------------------------------------------------------
+      logical function mpas_prev_ring_in_window(alarm)
+         implicit none
+         type(MPAS_Alarm_type), pointer :: alarm
+
+         mpas_prev_ring_in_window = mpas_time_in_interval( &
+               alarm%activeStartTime, alarm%activeStopTime, &
+               alarm%prevRingTime, is_open_interval=.true.)
```

Comment:
While I can see some benefits to having a generic routine `mpas_time_in_interval` and to using that routine here, at some point the use of such a routine may obscure the essence of what's going on in my opinion. What happens when we eventually want half-open intervals? Which arguments to `mpas_time_in_interval` are the interval endpoints, and which is the time to be compared against those endpoints?

Instead, what about simply using code like this (adjust whitespace to taste) to replace the above three lines?
```fortran
      mpas_prev_ring_in_window = (alarm % activeStartTime < alarm % prevRingTime &
                            .and. alarm % prevRingTime < alarm % activeStopTime)
```